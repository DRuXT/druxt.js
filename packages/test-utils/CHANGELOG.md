# druxt-test-utils

## 0.2.15

### Patch Changes

- Updated dependencies [41cab3a0]
  - druxt@0.22.0

## 0.2.14

### Patch Changes

- Updated dependencies [e46a3290]
  - druxt@0.21.0

## 0.2.13

### Patch Changes

- Updated dependencies [49b67872]
  - druxt@0.20.0

## 0.2.12

### Patch Changes

- Updated dependencies [be21952]
  - druxt@0.19.0

## 0.2.11

### Patch Changes

- Updated dependencies [e6cf1fc]
  - druxt@0.18.3

## 0.2.10

### Patch Changes

- 540afca: Updated drupal-jsonapi-params to 2.0.0
- Updated dependencies [540afca]
  - druxt@0.18.2

## 0.2.9

### Patch Changes

- Updated dependencies [80164a1]
  - druxt@0.18.1

## 0.2.8

### Patch Changes

- Updated dependencies [e3d5238]
  - druxt@0.18.0

## 0.2.7

### Patch Changes

- Updated dependencies [b79701c]
  - druxt@0.17.0

## 0.2.6

### Patch Changes

- Updated dependencies [7b749bd]
  - druxt@0.16.0

## 0.2.5

### Patch Changes

- Updated dependencies [2ae1d6d]
  - druxt@0.15.0

## 0.2.4

### Patch Changes

- Updated dependencies [45bc0b9]
  - druxt@0.14.0

## 0.2.3

### Patch Changes

- Updated dependencies [77ab204]
  - druxt@0.13.0

## 0.2.2

### Patch Changes

- Updated dependencies [4504a2f]
  - druxt@0.12.0

## 0.2.1

### Patch Changes

- Updated dependencies [75ff8a9]
  - druxt@0.11.0

## 0.2.0

### Patch Changes

- Updated dependencies [715e5ef]
  - druxt@0.10.0

## 0.1.0

### Patch Changes

- Updated dependencies [21170fb]
  - druxt@0.9.0
