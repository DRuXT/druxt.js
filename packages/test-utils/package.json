{
  "name": "druxt-test-utils",
  "version": "0.1.0",
  "description": "Test utilities for Druxt development.",
  "keywords": [
    "druxt",
    "jsonapi",
    "test"
  ],
  "exports": {
    ".": {
      "require": "./dist/druxt-test-utils.ssr.js",
      "import": "./dist/druxt-test-utils.esm.js"
    }
  },
  "main": "dist/druxt-test-utils.ssr.js",
  "module": "dist/druxt-test-utils.esm.js",
  "dependencies": {
    "drupal-jsonapi-params": "^1.2.2",
    "jest-mock-axios": "^4.4.0"
  },
  "peerDependencies": {
    "druxt": "*"
  }
}
