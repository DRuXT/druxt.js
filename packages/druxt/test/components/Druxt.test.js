import { createLocalVue, mount } from '@vue/test-utils'

import Druxt from '../../src/components/Druxt.vue'

// Setup local vue instance.
const localVue = createLocalVue()

const mountWrapper = (propsData) => {
  return mount(Druxt, { localVue, propsData, stubs: ['DruxtTestModule'] })
}

describe('Druxt component', () => {
  test('defaults', async () => {
    const wrapper = mountWrapper({ module: 'test-module' })

    // Props.
    expect(wrapper.vm.module).toBe('test-module')
    expect(wrapper.vm.propsData).toStrictEqual({})
    expect(wrapper.vm.wrapper.component).toBe('div')
    expect(wrapper.vm.inner.component).toBe('div')

    // Data.
    expect(wrapper.vm.component.is).toBe('DruxtTestModule')
    expect(wrapper.vm.component.propsData).toStrictEqual({})

    // HTML snapshot.
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('defaults - v-model', async () => {
    const Component = {
      template: "<Druxt v-model='model' ref='druxt' module='test-module' />",
      components: { Druxt },
      data: () => ({ model: null })
    }
    const wrapper = mount(Component, { localVue, stubs: ['DruxtTestModule'] })

    // Default state.
    expect(wrapper.vm.model).toStrictEqual(null)
    expect(wrapper.vm.$refs.druxt.model).toStrictEqual(null)
    expect(wrapper.vm.$refs.druxt.value).toStrictEqual(null)

    // Change model value.
    await wrapper.setData({ model: { test: true }})

    expect(wrapper.vm.model).toStrictEqual({ test: true })
    expect(wrapper.vm.$refs.druxt.model).toStrictEqual({ test: true })
    expect(wrapper.vm.$refs.druxt.value).toStrictEqual({ test: true })

    // Change Druxt component model value.
    wrapper.vm.$refs.druxt.model = { test: false }
    await localVue.nextTick()
    expect(wrapper.vm.model).toStrictEqual({ test: false })
    expect(wrapper.vm.$refs.druxt.model).toStrictEqual({ test: false })
    expect(wrapper.vm.$refs.druxt.value).toStrictEqual({ test: false })
  })

  test('missing module', async () => {
    const wrapper = mountWrapper({ module: 'test-missing-module' })
    expect(wrapper.vm.component.is).toBe(undefined)
  })

  test('wrapper', () => {
    const wrapper = mountWrapper({
      module: 'test-module',
      wrapper: {
        component: 'span'
      }
    })

    // Props.
    expect(wrapper.vm.wrapper.component).toBe('span')
    expect(wrapper.vm.inner.component).toBe('div')

    // HTML snapshot.
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('unwrapped', () => {
    const wrapper = mountWrapper({
      module: 'test-module',
      wrapper: false
    })

    // Props.
    expect(wrapper.vm.wrapper).toBe(false)
    expect(wrapper.vm.inner.component).toBe('div')

    // HTML snapshot.
    expect(wrapper.html()).toMatchSnapshot()
  })
})
