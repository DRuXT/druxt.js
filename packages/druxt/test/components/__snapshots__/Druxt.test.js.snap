// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Druxt component defaults 1`] = `
"<div>
  <druxttestmodule-stub></druxttestmodule-stub>
</div>"
`;

exports[`Druxt component unwrapped 1`] = `"<druxttestmodule-stub></druxttestmodule-stub>"`;

exports[`Druxt component wrapper 1`] = `"<span><druxttestmodule-stub></druxttestmodule-stub></span>"`;
